core/contracts/MultiplyGobblerVault.sol

```diff
@@ -166,50 +165,52 @@ contract MultiplyGobblerVault is ERC20, ERC721TokenReceiver, Owned, ReentrancyGu
 
     /// @notice Deposit Gobbler into the vault and get mGOB tokens proportional to multiplier of the Gobbler
     /// @dev This requires an approve before the deposit
-    /// @dev If GooDeposit is non-zero than take goo with Gobbler and also add it into Vault's virtual balance
+    /// @dev If GooDeposit is non-zero then take goo with Gobbler and also add it into Vault's virtual balance
     /// @param id id of the gobbler to mint
     function deposit(uint256 id) public {
         // multiplier of to be deposited gobbler
         uint256 multiplier = artGobbler.getGobblerEmissionMultiple(id);
-        // transfer art gobbler into the vault
-        artGobbler.safeTransferFrom(msg.sender, address(this), id);
-        // transfer go debt into the vault
         uint256 gooDeposit = getGooDeposit(multiplier);
         if (gooDeposit > 0) {
-            bool success = goo.transferFrom(msg.sender, address(this), gooDeposit);
-            if (!success) revert GooDepositFailed();
+            // transfer goo debt into the vault
+            // goo token reverts on invalid approval + insufficient balance
+            goo.transferFrom(msg.sender, address(this), gooDeposit);
             // adds any goo erc20 balance into the vaults virtual balance
             artGobbler.addGoo(goo.balanceOf(address(this)));
         }
+        // transfer art gobbler into the vault
+        artGobbler.transferFrom(msg.sender, address(this), id);
```

Comment:
why was this order changed? 
i.e. why in checks effects taking erc20 is preferred over erc721?

Reply:
No functional reason, just style. (Looking again, I'd probably put the transfer on L183 to be more consistent with the other functions). As long as nothing in `goo.transferFrom`, `artGobbler.addGoo`, or `artGobbler.transferFrom` changes state in a way that impacts the other functions, order doesn't matter.

I did forget to mention one other change here: 
- There's no need for `safeTransferFrom` on the ERC721 transfers, since this contract knows it can handle ERC721s.
- Goo/solmate ERC20 will revert on failed transfers. Although it returns a boolean, it will never be `false`, only `true` in the success case. Since you know this for sure, no need to check the return value.

Reply:
doing gods work ser!